### PR TITLE
Improve Actions status aggregations

### PR DESCRIPTION
**Pull request by wxiaoguang**
Make the result the same as GitHub:

* all skipped, then result is skipped
* any cancelled, then result cancelled

### CODE REVIEW

**Comment by wxiaoguang** [comment]
> Is Unknown a status which should change the final result? For example combination `Skipped, Success, Unknown` resolves currently to `Skipped` which isn't consistent with behavior of `Skipped, Success` combination.

Good catch, added more cases and comments in f438b563201d0bc40adfeb984ca92135bb9fd89d

**Review by TheFox0x7 — COMMENTED** [review]
LGTM
I'll check how Github treats all skipped workflows later.

File: models/actions/run_job_status_test.go [hunk 1]
```diff
@@ -38,18 +49,28 @@ func TestAggregateJobStatus(t *testing.T) {
...
 		{[]Status{StatusFailure, StatusSkipped}, StatusFailure},
-		{[]Status{StatusFailure, StatusCancelled}, StatusFailure},
+		{[]Status{StatusFailure, StatusCancelled}, StatusCancelled},
 		{[]Status{StatusFailure, StatusWaiting}, StatusFailure},
 		{[]Status{StatusFailure, StatusRunning}, StatusFailure},
 		{[]Status{StatusFailure, StatusBlocked}, StatusFailure},
 
 		// skipped with other status
-		{[]Status{StatusSkipped}, StatusSuccess},
+		// TODO: need to clarify whether a PR with "skipped" job status is considered as "mergeable" or not.
```
Comment (delvh) [inline models/actions/run_job_status_test.go:72]: I would say yes:
As I see it, a skipped job is mainly something like `execute on hardware xy, but it wasn't available`. Or perhaps something like `only execute once a month`. Still, that doesn't prevent the PR from merging.

**Review by delvh — APPROVED** [review]

**Review by Zettat123 — APPROVED** [review]

File: models/actions/run_job_status_test.go (hunk 1 above)
Comment (TheFox0x7) [reply models/actions/run_job_status_test.go:72]: Confirmed, skips are treated as a pass:
https://github.com/TheFox0x7/repro/pull/1
And cancellations as fails:
https://github.com/TheFox0x7/repro/pull/2/

Comment (wxiaoguang) [reply models/actions/run_job_status_test.go:72]: The "actions status to commit status" converting seems consistent:

https://github.com/go-gitea/gitea/blob/22c4599542ee3e10bcab4c9136467bbac8e90ba0/services/actions/commit_status.go#L143-L154

Comment (wxiaoguang) [reply models/actions/run_job_status_test.go:72]: Updated the comment in #32861 (with more trivial fixes)

**Pull request merged**